File: server/controllers/review_controller.py
from flask import request
from config.db_config import db
from models.models import Review, Vendor
from utils.helpers import success_response, error_response

def get_vendor_reviews(vendor_id):
    reviews = Review.query.filter_by(vendor_id=vendor_id).order_by(Review.created_at.desc()).all()
    return success_response([r.to_dict() for r in reviews])

def create_review(user_id):
    data = request.get_json() or {}
    vendor_id = data.get('vendor_id') or data.get('vendorId')
    rating = data.get('rating')
    comment = data.get('comment', '')

    if not vendor_id or rating is None:
        return error_response('Vendor ID and rating are required.', 400)

    vendor = Vendor.query.get(vendor_id)
    if not vendor:
        return error_response('Vendor not found.', 404)

    review = Review(
        user_id=user_id,
        vendor_id=vendor_id,
        rating=int(rating),
        comment=comment
    )
    db.session.add(review)

    # Recalculate vendor average rating
    all_reviews = Review.query.filter_by(vendor_id=vendor_id).all()
    total_ratings = sum(r.rating for r in all_reviews) + int(rating)
    count = len(all_reviews) + 1
    vendor.rating = round(total_ratings / count, 1)
    vendor.reviews_count = count

    db.session.commit()
    return success_response(review.to_dict(), 'Review submitted successfully.', 201)

File: server/controllers/reviewController.js
const { query } = require('../config/db');

const getVendorReviews = async (req, res, next) => {
  try {
    const vendorId = Number(req.params.vendorId);
    let reviews = [];
    try {
      const [dbReviews] = await query('SELECT * FROM reviews WHERE vendor_id = ? ORDER BY created_at DESC', [vendorId]);
      reviews = dbReviews || [];
    } catch (e) {
      reviews = [];
    }

    res.json({
      success: true,
      count: reviews.length,
      reviews,
      data: {
        reviews
      }
    });
  } catch (err) {
    next(err);
  }
};

const createReview = async (req, res, next) => {
  try {
    const payload = req.body || {};
    const vendorId = payload.vendorId || payload.vendor_id;
    const reviewerName = payload.reviewerName || payload.reviewer_name || (req.user ? req.user.name : 'Verified Customer');
    const rating = payload.rating;
    const comment = payload.comment || payload.review;
    const eventType = payload.eventType || payload.event_type || 'Wedding';

    if (!vendorId || !rating || !comment) {
      return res.status(400).json({
        success: false,
        message: 'Vendor ID, rating (1-5), and comment are required.'
      });
    }

    const userId = req.user ? req.user.id : null;

    let reviewId = 10;
    try {
      const insertSql = `
        INSERT INTO reviews (vendor_id, user_id, reviewer_name, rating, comment, event_type)
        VALUES (?, ?, ?, ?, ?, ?)
      `;

      const result = await query(insertSql, [
        Number(vendorId),
        userId,
        reviewerName,
        Number(rating),
        comment,
        eventType
      ]);
      reviewId = result[0]?.insertId || 10;
    } catch (e) {
      console.warn('[EventHub DB] Insert review fallback:', e.message);
    }

    const newReview = {
      id: reviewId,
      vendor_id: Number(vendorId),
      vendorId: Number(vendorId),
      reviewer_name: reviewerName,
      reviewerName,
      rating: Number(rating),
      comment,
      event_type: eventType,
      eventType,
      created_at: new Date().toISOString()
    };

    res.status(201).json({
      success: true,
      message: 'Review posted successfully.',
      review: newReview,
      data: newReview,
      ...newReview
    });
  } catch (err) {
    next(err);
  }
};

module.exports = {
  getVendorReviews,
  createReview
};
